template/src/ontology/Makefile.jinja2

```diff
@@ -946,12 +948,13 @@ update_docs:
 	mkdocs gh-deploy --config-file ../../mkdocs.yaml
 {%- endif %}
 
+# Note to future generations: prepending ./ is a safety measure to ensure that 
+# the environment does not malicously set `CLEANFILES` to `\`.
 .PHONY: clean
 clean:{% if project.use_dosdps %}
 	make pattern_clean{%- endif %}
-	rm -rf $(TMPDIR)/
-	rm -rf $(MIRROR)/
-	rm -f $(CLEANFILES)
+	rm -rf ./tmp/* ./mirror/*
+	rm -f $(patsubst %, ./%, $(CLEANFILES))
```

Comment:
cc @gouttegd

Reply:
OK, but this may break the user’s expectation that they are allowed to redefine `MIRRORDIR` or `TMPDIR` in the custom Makefile, and have the entire Makefile take this redefinition into account.

I’d prefer to keep referring to the mirror and temporary directories using `MIRRORDIR` and `TMPDIR`, with possibly some checks before `rm`-ing them if you are concerned they could have been re-defined in some harmful way.

E.g., check that the variable is not empty and does point to a directory existing in the current folder:

```
[ -n "$(MIRRORDIR)" ] && [ -d ./$(MIRRORDIR) ] && rm -rf ./$(MIRRORDIR)/*
```

Depending on your paranoia level, you may also want to protect against the case where `MIRRORDIR` is maliciously set to `.` or `..`:

```
[ -n "$(MIRRORDIR)" ] && [ $(MIRRORDIR) != "." ] && [ $(MIRRORDIR) != ".." ] && [ -d ./$(MIRRORDIR) ] && rm -rf ./$(MIRRORDIR)/*
```

Note that this does not protect against the case where `MIRRORDIR` is even more malicious, e.g. `MIRRORDIR=normal/path/../../../../oh/noes`). But I am not convinced  we should go into a mode where we systematically assume someone _will_ try to play tricks with us (we could go into a rabbit hole if we start thinking like that).

Reply:
Thanks a ton man, I did what you suggested. I will ask you during the call later to review one last time.